src/Pos/Delegation/Logic.hs

```diff
@@ -480,15 +480,19 @@ processProxySKHeavy psk = do
     maxBlockSize <- bvdMaxBlockSize <$> DB.gsAdoptedBVData
     let msg = Right psk
         consistent = verifyProxySecretKey psk
-        issuer = pskIssuerPk psk
-        enoughStake = addressHash issuer `elem` richmen
+        iPk = pskIssuerPk psk
+        -- We don't check stake for revoking certs. You can revoke
+        -- even if you don't have money anymore.
+        enoughStake = isRevokePsk psk || addressHash iPk `elem` richmen
         omegaCorrect = headEpoch == pskOmega psk
     runDelegationStateAction $ do
         memPoolSize <- use dwPoolSize
-        exists <- uses dwProxySKPool (\m -> HM.lookup issuer m == Just psk)
+        exists <- uses dwProxySKPool (\m -> HM.lookup iPk m == Just psk)
         cached <- isJust . snd . LRU.lookup msg <$> use dwMessageCache
-        alreadyPosted <- uses dwThisEpochPosted $ HS.member issuer
+        alreadyPosted <- uses dwThisEpochPosted $ HS.member iPk
         epochMatches <- (headEpoch ==) <$> use dwEpochId
+        hasPskInDB <- fmap isJust $ GS.getPskByIssuer (Left $ pskIssuerPk psk)
+        let rerevoke = isRevokePsk psk && hasPskInDB
```

Comment:
I don't get this logic, why is it rerevokation?
Seems, when you just revoke the certificate, this statement (`isRevokePsk psk && hasPskInDB`) also will true.